webapi/webapi-serviceworkers-w3c-tests/serviceworkers/w3c/cache-storage/script-tests/cache-add.js

```diff
@@ -0,0 +1,135 @@
+        });
+  }, 'Cache.add with request that results in a status of 404');
+
+cache_test(function(cache) {
+    return cache.add('../resources/fetch-status.py?status=500')
+      .then(function(result) {
+          assert_equals(result, undefined,
+                        'Cache.add should resolve with undefined on success.');
+        });
+  }, 'Cache.add with request that results in a status of 500');
+
+cache_test(function(cache) {
+    return assert_promise_rejects(
+      cache.addAll(),
+      new TypeError(),
+      'Cache.addAll with no arguments should throw TypeError.');
+  }, 'Cache.addAll with no arguments');
+
+cache_test(function(cache) {
+    // Assumes the existence of ../resources/simple.txt and ../resources/blank.html
+    var urls = ['../resources/simple.txt', undefined, '../resources/blank.html'];
+    return assert_promise_rejects(
+      cache.addAll(),
+      new TypeError(),
+      'Cache.addAll should throw TypeError for an undefined argument.');
+  }, 'Cache.addAll with a mix of valid and undefined arguments');
+
+cache_test(function(cache) {
+    // Assumes the existence of ../resources/simple.txt and ../resources/blank.html
+    var urls = ['../resources/simple.txt', self.location.href, '../resources/blank.html'];
+    return cache.addAll(urls)
+      .then(function(result) {
+          assert_equals(result, undefined,
+                        'Cache.addAll should resolve with undefined on ' +
+                        'success.');
+        });
+  }, 'Cache.addAll with string URL arguments');
+
+cache_test(function(cache) {
+    // Assumes the existence of ../resources/simple.txt and ../resources/blank.html
+    var urls = ['../resources/simple.txt', self.location.href, '../resources/blank.html'];
+    var requests = urls.map(function(url) {
+        return new Request(url);
+      });
+    return cache.addAll(requests)
+      .then(function(result) {
+          assert_equals(result, undefined,
+                        'Cache.addAll should resolve with undefined on ' +
+                        'success.');
+        });
+  }, 'Cache.addAll with Request arguments');
+
+cache_test(function(cache) {
+    // Assumes that ../resources/simple.txt and ../resources/blank.html exist. The second
+    // resource does not.
+    var urls = ['../resources/simple.txt', 'this-resource-should-not-exist', '../resources/blank.html'];
+    var requests = urls.map(function(url) {
+        return new Request(url);
+      });
+    return cache.addAll(requests)
+      .then(function(result) {
+          assert_equals(result, undefined,
+                        'Cache.addAll should resolve with undefined on ' +
+                        'success.');
+        });
+  }, 'Cache.addAll with a mix of succeeding and failing requests');
+
+cache_test(function(cache) {
+    var request = new Request('../resources/simple.txt');
+    return assert_promise_rejects(
+      cache.addAll([request, request]),
+      new TypeError(),//'InvalidStateError',
```

Comment:
You changed this line of code, why?

Why not submit a PR against this change to W3C w-p-t repo?